### PR TITLE
Note simulation

### PR DESCRIPTION
**Pull request by YoniKiriaty**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 1]
```diff
@@ -47,6 +47,7 @@ private OI() {
 		fourthJoystick = new SmartJoystick(RobotConstants.Joystick.FOURTH);
 		
 		initButtons();
+		thirdJoystickButtons();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/OI.java:50]: delete

File: src/main/java/edu/greenblitz/robotName/commands/shooter/PushNoteToFlyWheel.java [hunk 2]
```diff
@@ -12,7 +13,7 @@ public PushNoteToFlyWheel() {
         super(
                 new ForwardRunFunnelUntilObjectOut(),
                 new WaitCommand(FlyWheelConstants.DELAY_SECONDS_UNTIL_EXIT),
-                new StopFlyWheel()
+                new StopFlyWheel().alongWith(new ShootSimulationNote())
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/shooter/PushNoteToFlyWheel.java:16]: shouldnt be here

File: src/main/java/edu/greenblitz/robotName/commands/shooter/flyWheel/ShootSimulationNote.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        ).times(timer.get() * FlyWheelConstants.SIMULATION_SHOOTING_SPEED_METERS_PER_SECOND);
+
+        Logger.recordOutput(
+                "NoteVisualizer",
+                new Pose3d[]{
+                        new Pose3d(
+                                notePosition.plus(shooterPose3D.getTranslation()),
+                                Pivot.getInstance().getSimulationPivotPosition3d().getRotation().plus(shooterPose3D.getRotation())
+                        )
+                });
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/commands/shooter/flyWheel/ShootSimulationNote.java:53]:
```suggestion
                }
        );
```

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/chassis/SwerveChassis.java [hunk 4]
```diff
@@ -409,15 +410,23 @@ public boolean isRobotOnGround() {
...
-    public Pose2d getRobotPose() {
+    public Pose2d getRobotPose2d() {
         return poseEstimator.getEstimatedPosition();
     }
 
+    public Pose3d getRobotPose3d() {
+        Pose2d swervePose2d = getRobotPose2d();
+        return new Pose3d(
+                new Translation3d(swervePose2d.getX(), swervePose2d.getY(), RobotConstants.SimulationConstants.ROBOT_TRANSLATION.getZ()),
+                new Rotation3d(0,0,swervePose2d.getRotation().getRadians())
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/chassis/SwerveChassis.java:426]: add spaces

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/chassis/SwerveChassis.java [hunk 5]
```diff
@@ -409,15 +410,23 @@ public boolean isRobotOnGround() {
...
 
-    public Pose2d getRobotPose() {
+    public Pose2d getRobotPose2d() {
         return poseEstimator.getEstimatedPosition();
     }
 
+    public Pose3d getRobotPose3d() {
+        Pose2d swervePose2d = getRobotPose2d();
+        return new Pose3d(
+                new Translation3d(swervePose2d.getX(), swervePose2d.getY(), RobotConstants.SimulationConstants.ROBOT_TRANSLATION.getZ()),
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/chassis/SwerveChassis.java:425]: spaces

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/utils/GBPathFinding/GBAutoBuilder.java
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/utils/GBPathFinding/GBAutoBuilder.java:1]: Empty file?

File: src/main/java/edu/greenblitz/robotName/OI.java [hunk 6]
```diff
@@ -93,9 +93,9 @@ public void initButtons() {
 	public void thirdJoystickButtons() {
 		SmartJoystick usedJoystick = thirdJoystick;
 		usedJoystick.A.whileTrue(new RunRollerByJoystick(usedJoystick));
-		usedJoystick.A.whileTrue(new RunIntakeByJoystick(usedJoystick));
-		usedJoystick.B.whileTrue(new RunFunnelByJoystick(usedJoystick));
-		usedJoystick.B.whileTrue(new RunFlyWheelByJoystick(usedJoystick));
+		usedJoystick.B.whileTrue(new RunIntakeByJoystick(usedJoystick));
+		usedJoystick.X.whileTrue(new RunFunnelByJoystick(usedJoystick));
+		usedJoystick.Y.whileTrue(new ShootSimulationNote());
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/OI.java:98]: Use 4th controller

File: src/main/java/edu/greenblitz/robotName/RobotConstants.java [hunk 7]
```diff
@@ -58,7 +58,7 @@ public static class SimulationConstants {
 		
 		public static final double MIN_MOTOR_VOLTAGE = -12;
 		
-		public static final Translation3d ARM_TO_ROBOT = new Translation3d(0, 0, 1.2);
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/RobotConstants.java:61]: Is this not being used?

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheelConstants.java [hunk 8]
```diff
@@ -8,7 +8,7 @@ public class FlyWheelConstants {
 	
 	public static final double EPSILON_RPM = 50;
 	
-	public static final double SHOOTING_SPEED_TIME = 1000;
+	public static final double SHOOTING_SPEED_TIME = -1;
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/FlyWheel/FlyWheelConstants.java:11]: What? Also add units to var name

File: src/main/java/edu/greenblitz/robotName/utils/GBPathFinding/GBAutoBuilder.java
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/utils/GBPathFinding/GBAutoBuilder.java:1]: This file is empty

**Review by yonnbro — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/RobotConstants.java (hunk 7 above)
Comment (YoniKiriaty) [reply src/main/java/edu/greenblitz/robotName/RobotConstants.java:61]: No it is not

**Review by yonnbro — APPROVED** [review]

**Pull request merged**